opentracing/src/main/java/opentracing/SpanPropagator.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+import java.util.Map;
+
+/**
+ * Encodes or Decodes a {@linkplain Span} in binary or text formats.
+ *
+ * <p>The toXXX methods are expected to serialize spans into a pair of values representing
+ * separately the span / span identity, and the trace attributes. This is done specifically
+ * for binary protocols that may represent tracing identity in a dedicated fixed-length slot in the
+ * binary message format, so that it can be inspected efficiently by the middleware / routing layers
+ * without parsing the whole message.
+ */
+public interface SpanPropagator {
+
+  /**
+   * Encodes the span into a binary representation of the span's identity and trace
+   * attributes.
+   */
+  byte[] toBinary(Span span);
```

Comment:
I should clarify slightly.. Zipkin backends don't require or have an opinion on extra propagated things. Individual tracers may or may not.

Traditionally, zipkin tracers have not intentionally exposed their DCP apis to users directly. They do use common header names for interop, but for example don't specificy a naming convention or otherwise to carry other data.